test/files/coverageparser.test.ts

```diff
@@ -11,41 +20,29 @@ suite("CoverageParser Tests", function() {
         testFiles.set("/file/222", "datastring222");
         testFiles.set("/file/123", "samestringhere");
 
-        // Mock lcovExtract
-        const lcovExtract = async (filename) => {
+        const coverageParsers = new CoverageParser(fakeOutputChannel);
+        sinon.stub(coverageParsers as any, "lcovExtract").callsFake(async (filename) => {
```

Comment:
[F] There were two test that I couldn't find away around casting the object to any so that we could stub a private function. This one and `"findCoverageFiles returns an error if no coverage file @unit"`. I think that it is reasonable to leave these as is for now.

Reply:
Sounds good, we could also use sinon to make the fakes instead of making our own on line 11 but otherwise this makes sense!

Reply:
I'll have another look at this but I think that I got a type error when trying to create an output channel stub with sinon.

Reply:
When trying to create a stub using `sinon.createStubInstance(OutputChannel) as OutputChannel;` the error is `'OutputChannel' only refers to a type, but is being used as a value here.` which will be because we don't have access to the constructor or class. I could stub `vscode.window.createOutputChannel` and have it return our fake object but I don't think it is worth the effort. WDYT @ryanluker?

Reply:
Yah that is proving tricky, I think this is good enough.